server/runtime/src/main/java/org/infinispan/server/configuration/hotrod/HotRodServerConfigurationParser.java

```diff
@@ -123,6 +128,50 @@ private void parseHotRodConnector(XMLExtendedStreamReader reader, ConfigurationB
             }
          }
       }
+      if (implicitAuthentication) {
+         ServerSecurityRealm securityRealm = serverBuilder.endpoint().securityRealm();
+         builder.authentication().enable().securityRealm(securityRealm.getName());
+         String serverPrincipal = null;
+         for (KerberosSecurityFactoryConfiguration identity : securityRealm.getServerIdentities().kerberosConfigurations()) {
+            if (identity.getPrincipal().startsWith("hotrod/")) {
+               builder.authentication()
+                     .addMechanisms(SaslMechanismInformation.Names.GS2_KRB5, SaslMechanismInformation.Names.GSSAPI);
+               serverPrincipal = identity.getPrincipal();
+               break;
+            }
+            Server.log.debugf("Enabled Kerberos mechanisms for Hot Rod using principal '%s'", identity.getPrincipal());
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TOKEN)) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.OAUTHBEARER);
+            Server.log.debug("Enabled OAUTHBEARER mechanism for Hot Rod");
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TRUST)) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.EXTERNAL);
+            Server.log.debug("Enabled EXTERNAL mechanism for Hot Rod");
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.PASSWORD)) {
+            builder.authentication()
+                  .addMechanisms(
+                        SaslMechanismInformation.Names.SCRAM_SHA_512,
+                        SaslMechanismInformation.Names.SCRAM_SHA_384,
+                        SaslMechanismInformation.Names.SCRAM_SHA_256,
+                        SaslMechanismInformation.Names.SCRAM_SHA_1,
+                        SaslMechanismInformation.Names.DIGEST_SHA_512,
+                        SaslMechanismInformation.Names.DIGEST_SHA_384,
+                        SaslMechanismInformation.Names.DIGEST_SHA_256,
+                        SaslMechanismInformation.Names.DIGEST_SHA,
+                        SaslMechanismInformation.Names.CRAM_MD5,
+                        SaslMechanismInformation.Names.DIGEST_MD5
+                  );
+            Server.log.debug("Enabled SCRAM, DIGEST and CRAM mechanisms for Hot Rod");
+         }
+         // Only enable PLAIN if encryption is on
+         if (serverBuilder.endpoint().ssl().isEnabled()) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.PLAIN);
+            Server.log.debug("Enabled PLAIN mechanism for Hot Rod");
```

Comment:
I would add the comment in the debug. Encryption is on. PLAIN mechanism enabled for Hot Rod

Reply:
If we see this log, it already means that.